File: GXpert/GXpert.Web/Modules/Masters/PreAcademicScore/PreAcademicScoreForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Masters.Forms;

[FormScript("Masters.PreAcademicScore")]
[BasedOnRow(typeof(PreAcademicScoreRow), CheckNames = true)]
public class PreAcademicScoreForm
{
    [HalfWidth]
    public int PreAcadamicsId { get; set; }
    [HalfWidth]
    public DateTime PassedOutDate { get; set; }
    [HalfWidth]
    public float MarksObtained { get; set; }
    [HalfWidth]
    public float OutOfMarks { get; set; }
    [HalfWidth]
    public int StudentId { get; set; }
    [HalfWidth]
    public string Remarks { get; set; }
  
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Masters/PreAcademicScoreForm.ts
import { LookupEditor, DateEditor, DecimalEditor, StringEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface PreAcademicScoreForm {
    PreAcadamicsId: LookupEditor;
    PassedOutDate: DateEditor;
    MarksObtained: DecimalEditor;
    OutOfMarks: DecimalEditor;
    StudentId: LookupEditor;
    Remarks: StringEditor;
}

export class PreAcademicScoreForm extends PrefixedContext {
    static readonly formKey = 'Masters.PreAcademicScore';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PreAcademicScoreForm.init)  {
            PreAcademicScoreForm.init = true;

            var w0 = LookupEditor;
            var w1 = DateEditor;
            var w2 = DecimalEditor;
            var w3 = StringEditor;

            initFormType(PreAcademicScoreForm, [
                'PreAcadamicsId', w0,
                'PassedOutDate', w1,
                'MarksObtained', w2,
                'OutOfMarks', w2,
                'StudentId', w0,
                'Remarks', w3
            ]);
        }
    }
}